int32 agent_id
bool is_leader 